music/src/music_vae/model.ts

```diff
@@ -669,6 +669,21 @@ export interface MusicVAESpec {
   useBooleanDecoder: boolean;
 }
 
+/**
+ * Interface for conditioning arguments to MusicVAE functions.
+ *
+ * @property chordProgression (Optional) Chord progression, an array of chord
+ * symbol strings like "Cmaj7", "Abdim", "C#m7", etc.
+ * @property key (Optional) Key, an integer between 0 (C major / A minor) and 11
+ * (B major / G# minor).
+ * @property extraControls (Optional) Additional control tensors.
+ */
+export interface MusicVAEControlArgs {
+  chordProgression?: string[];
+  key?: number;
+  extraControls?: tf.Tensor2D;
```

Comment:
Just curious why `extraControls` is simply a concatenated `Tensor2D` of rhythm/contour and not specific a key/value for each of those?

Reply:
Do you mean why is `extraControls` a `Tensor2D` instead of a dict of `name` -> `Tensor2D`?  And then the model config can specify which controls are needed?  That's not a bad idea...

Reply:
@adarob What do you think about this?

Reply:
I like @cannoneyed's suggestion if it's easy to do.

Reply:
But also, why are these tensors not just separate fields in the interface?

Reply:
I think I don't want to define controls like rhythm or shape contour here as they are (as we've been using them) specific to melody.  But I can make `extraControls` a dict.

Reply:
Done, and updated checkpoints to work with the new (and old) code.